### ISSUE TITLE
ERROR: Your RubyGems was installed trough APT, and upgrading it through RubyGems itself is unsupported.

### ISSUE THREAD

**Issue by thundeRishi** [beefproject/beef#2441]
ERROR:  Your RubyGems was installed trough APT, and upgrading it through RubyGems
itself is unsupported. If you really need the latest version of RubyGems (tip:
you usually don't), then you need to install RubyGems (and Ruby) manually,
maybe using tools like ruby-install, rvm, etc.

i know i have seen a lot of solution but none of them worked please help

**Comment by jcrew99** [comment]
This is an installation error and it depends on the context of how rubygems was initially installed.
Can you confirm if this is happening when you run the ./install file?

A lot of questions come up, what is your rubygems version, what were you doing when it was thrown and why are you trying to upgrade it?

**Comment by bcoles** [comment]
> i know i have seen a lot of solution but none of them worked please help

https://github.com/beefproject/beef/issues?q=is%3Aissue+%22your+RubyGems+was+installed+trough+APT%22+is%3Aclosed

Where did you get stuck during troubleshooting?

You should be able to do this:

```
cd /path/to/beef
gem install bundler
bundle install
```

BeEF has not yet been fully updated to Ruby 3. If you're using Ruby 3, try:

```
cd /path/to/beef
rm Gemfile.lock
bundle3.0 install
```

**Comment by jcrew99** [comment]
Have you had any success during further troubleshooting?

**Comment by thundeRishi** [comment]
Can you tell me from scrath how to install beef

**Comment by thundeRishi** [comment]
i am going to deleat each and every file

**Comment by bcoles** [comment]
> Can you tell me from scrath how to install beef

Install Ruby, install Ruby gems, install `bundle` gem, `bundle install`.

https://github.com/beefproject/beef/issues/2441#issuecomment-1145547256

**Comment by bcoles** [comment]
This should be fixed in #2612. Failure to upgrade system rubygems now results in a warning rather than fatal.
